### PR TITLE
Link Migration Backend

### PR DESCRIPTION
**Pull request by tacryt-socryp**
TODO:

- [x]  wipe all subscriptions in link-proxy and link-listen hooks

Edit: Should be good to go!

### CODE REVIEW

File: pkg/arvo/app/link-view.hoon [hunk 1]
```diff
@@ -1,626 +1,40 @@
...
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+::
+++  on-init  [~ this]
+++  on-save  !>(state)
+++  on-load
+  |=  old-vase=vase
+  ^-  (quip card _this)
+  :_  this(state [%2 ~])
+  ::  TODO: leave all subscriptions
```
Comment (belisarius222) [inline pkg/arvo/app/link-view.hoon:30]: Is this TODO still relevant?

File: pkg/arvo/mar/graph/validator/link.hoon [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ++  noun  i
+  --
+++  grab
+  |%
+  ++  noun
+    |=  p=*
+    =/  ip  ;;(indexed-post p)
+    ?+    index.p.ip  ~|(index+index.p.ip !!)
+        ::  top-level link post; title and url
+        ::
```
Comment (belisarius222) [inline pkg/arvo/mar/graph/validator/link.hoon:14]: looks like a stray unicode character

**Review by belisarius222 — COMMENTED** [review]
LGTM other than that possibly relevant TODO

File: pkg/arvo/app/link-view.hoon (hunk 1 above)
Comment (tacryt-socryp) [reply pkg/arvo/app/link-view.hoon]: Not particularly.

**Review by belisarius222 — APPROVED** [review]
LGTM

**Comment by galenwp** [comment]
excited for this to go in

File: pkg/arvo/app/graph-push-hook.hoon [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+++  on-leave  on-leave:def
+++  on-peek   on-peek:def
+++  on-arvo   on-arvo:def
+++  on-fail   on-fail:def
+::
+++  should-proxy-update
+  |=  =vase
+  ^-  ?
+  =/  =update:store  !<(update:store vase)
+  ?-  -.q.update
```
Comment (liam-fitzgerald) [inline pkg/arvo/app/graph-push-hook.hoon:48]: Ought to check the relevant group for membership

File: pkg/arvo/app/graph-push-hook.hoon [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      %keys               ~
+      %tags               ~
+      %tag-queries        ~
+      %run-updates        `resource.q.update
+  ==
+::
+++  initial-watch
+  |=  [=path =resource:res]
+  ^-  vase
+  ?>  (can-join:grp resource src.bowl)
```
Comment (liam-fitzgerald) [inline pkg/arvo/app/graph-push-hook.hoon:89]: realistically, we should check membership here, not if they can join.

Comment (liam-fitzgerald) [reply pkg/arvo/app/graph-push-hook.hoon:89]: Also needs to look up the group in the md-store.

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
Some questions surrounding permissioning of graphs. Otherwise, looks good

File: pkg/arvo/app/graph-push-hook.hoon (hunk 4 above)
Comment (tacryt-socryp) [reply pkg/arvo/app/graph-push-hook.hoon]: I've updated it. Does this mechanism of looking up the group in the `metadata-store` look alright?

**Review by belisarius222 — APPROVED** [review]
LGTM

**Pull request merged**